python/lsst/analysis/tools/interfaces/_stages.py

```diff
@@ -223,7 +223,12 @@ def __call__(self, data: KeyedData, **kwargs) -> MetricResultType:
             if newName := self.newNames.get(key):
                 formattedKey = newName.format(**kwargs)
             notes = {"metric_tags": kwargs.get("metric_tags", [])}
-            results[formattedKey] = Measurement(formattedKey, value * apu.Unit(unit), notes=notes)
+            if isinstance(value, dict):
```

Comment:
I'm tempted to suggest making this more generic, and checking `if isinstance(value, Mapping):`. Is that something that might be useful down the line?

Reply:
It's probably best not to limit ourselves to dictionaries. I tested it, and it works.